streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -678,6 +678,114 @@ trait ZSink[-R, +E, +A0, -A, +B] { self =>
         }
     }
 
+  final def zipPar[R1 <: R, E1 >: E, A2 >: A0, A1 <: A, C](
+    that: ZSink[R1, E1, A2, A1, C]
+  ): ZSink[R1, E1, A2, A1, (B, C)] =
+    new ZSink[R1, E1, A2, A1, (B, C)] {
+      type State = (Either[B, self.State], Either[C, that.State])
+
+      override def extract(state: State): ZIO[R1, E1, (B, C)] = {
+        val b: ZIO[R, E, B]   = state._1.fold(ZIO.succeed, s1 => self.extract(s1))
+        val c: ZIO[R1, E1, C] = state._2.fold(ZIO.succeed, s2 => that.extract(s2))
+        b.zip(c)
+      }
+
+      override def initial: ZIO[R1, E1, Step[State, Nothing]] =
+        self.initial.flatMap(s1 => {
+          that.initial.flatMap(s2 => {
+            (Step.cont(s1), Step.cont(s2)) match {
+              case (false, false) => {
+                val zb = self.extract(Step.state(s1))
+                val zc = that.extract(Step.state(s2))
+                zb.zipWith(zc)((b, c) => Step.done((Left(b), Left(c)), Chunk.empty))
+              }
+
+              case (false, true) => {
+                val zb = self.extract(Step.state(s1))
+                zb.map(b => Step.more((Left(b), Right(Step.state(s2)))))
+              }
+
+              case (true, false) => {
+                val zc = that.extract(Step.state(s2))
+                zc.map(c => Step.more((Right(Step.state(s1)), Left(c))))
+              }
+
+              case (true, true) => {
+                ZIO.succeed(Step.more((Right(Step.state(s1)), Right(Step.state(s2)))))
+              }
+            }
+          })
+        })
+
+      override def step(state: State, a: A1): ZIO[R1, E1, Step[State, A2]] = {
+        val firstResult: ZIO[R, E, Either[(B, Option[Chunk[A2]]), self.State]] = state._1.fold(
+          b => ZIO.succeed(Left((b, None))),
+          s =>
+            self
+              .step(s, a)
+              .flatMap(((st: Step[ZSink.this.State, A0]) => {
+                if (Step.cont(st)) {
+                  ZIO.succeed(Right(Step.state(st)))
+                } else {
+                  self
+                    .extract(Step.state(st))
+                    .map(b => {
+                      Left((b, Some(Step.leftover(st))))
+                    })
+                }
+              }))
+        )
+
+        val secondResult: ZIO[R1, E1, Either[(C, Option[Chunk[A2]]), that.State]] = state._2.fold(
+          c => ZIO.succeed(Left((c, None))),
+          s =>
+            that
+              .step(s, a)
+              .flatMap(st => {
+                if (Step.cont(st)) {
```

Comment:
When your `if/else` block body consists of just one expression, you could drop the braces around it as well